build-system/tasks/prepend-global/index.js

```diff
@@ -136,26 +137,42 @@ function applyConfig(
   return checkoutBranchConfigs(filename, opt_localBranch, opt_branch)
     .then(() => {
       return Promise.all([
-        fs.promises.readFile(filename),
-        fs.promises.readFile(target),
+        fs.promises.readFile(filename, 'utf8'),
+        fs.promises.readFile(target, 'utf8'),
+        fs.promises.readFile(customConfigFile, 'utf8').catch(() => {}),
       ]);
     })
     .then((files) => {
+      let configString = files[0];
```

Comment:
nit that other reviewers may object to: `then(([configString, targetString, overlayString]) => {`

Reply:
No objection to the syntax suggested here. In fact, I like it.

Side note: The code in this file is quite ancient. I've meant to replace all the `.then()` blocks with `async` and `await`, but never got down to doing it. Will do sometime!

Reply:
Done.